opampserver/pkg/server/handlers.go

```diff
@@ -50,56 +55,73 @@ func (c *ConnectionHandlers) OnNewConnection(ctx context.Context, deviceId strin
 	}
 
+	if firstMessage.PackageStatuses == nil {
+		// this might come from an older agent which not yet supports package statuses
+		c.logger.Info("missing package statuses in first agent to server message", "deviceId", deviceId)
+	}
+
 	k8sAttributes, pod, err := c.deviceIdCache.GetAttributesFromDevice(ctx, deviceId)
 	if err != nil {
 		c.logger.Error(err, "failed to get attributes from device", "deviceId", deviceId)
 		return nil, nil, err
 	}
-	serverOfferedResourceAttributes, err := calculateServerAttributes(k8sAttributes, c.nodeName)
-	if err != nil {
-		c.logger.Error(err, "failed to calculate server attributes", "deviceId", deviceId)
-		return nil, nil, err
+
+	podWorkload := common.PodWorkload{
+		Namespace: pod.GetNamespace(),
+		Kind:      k8sAttributes.WorkloadKind,
+		Name:      k8sAttributes.WorkloadName,
 	}
 
 	instrumentedAppName := workload.GetRuntimeObjectName(k8sAttributes.WorkloadName, k8sAttributes.WorkloadKind)
-	c.logger.Info("new OpAMP client connected", "deviceId", deviceId, "namespace", k8sAttributes.Namespace, "podName", k8sAttributes.PodName, "instrumentedAppName", instrumentedAppName, "workloadKind", k8sAttributes.WorkloadKind, "workloadName", k8sAttributes.WorkloadName, "containerName", k8sAttributes.ContainerName, "otelServiceName", k8sAttributes.OtelServiceName)
+	remoteResourceAttributes, err := configresolvers.CalculateServerAttributes(k8sAttributes, c.nodeName)
+	if err != nil {
+		c.logger.Error(err, "failed to calculate server attributes", "k8sAttributes", k8sAttributes)
+		return nil, nil, err
+	}
 
-	sdkConfig, err := json.Marshal(RemoteConfigSdk{RemoteResourceAttributes: serverOfferedResourceAttributes})
+	fullRemoteConfig, err := c.sdkConfig.GetFullConfig(ctx, remoteResourceAttributes, &podWorkload)
 	if err != nil {
-		c.logger.Error(err, "failed to marshal server offered resource attributes")
+		c.logger.Error(err, "failed to get full config", "k8sAttributes", k8sAttributes)
 		return nil, nil, err
 	}
 
-	serverAttrsRemoteCfg := protobufs.AgentRemoteConfig{
-		Config: &protobufs.AgentConfigMap{
-			ConfigMap: map[string]*protobufs.AgentConfigFile{
-				string(RemoteConfigSdkConfigSectionName): {
-					Body:        sdkConfig,
-					ContentType: "application/json",
-				},
-			},
-		},
+	c.logger.Info("new OpAMP client connected", "deviceId", deviceId, "namespace", k8sAttributes.Namespace, "podName", k8sAttributes.PodName, "instrumentedAppName", instrumentedAppName, "workloadKind", k8sAttributes.WorkloadKind, "workloadName", k8sAttributes.WorkloadName, "containerName", k8sAttributes.ContainerName, "otelServiceName", k8sAttributes.OtelServiceName)
+
+	connectionInfo := &connection.ConnectionInfo{
+		DeviceId:                 deviceId,
+		Workload:                 podWorkload,
+		Pod:                      pod,
+		Pid:                      pid,
+		InstrumentedAppName:      instrumentedAppName,
+		AgentRemoteConfig:        fullRemoteConfig,
+		RemoteResourceAttributes: remoteResourceAttributes,
```

Comment:
as i understood it, the `remoteResourceAttributes` is part of the fullRemoteConfig, why we need it in the ConnectionInfo as well?
I saw that we save it in the connectionCache and never use it from the cache

Reply:
We use it when one of the other reconcilers updates and we need to calculate the entire full config from scratch. If we don't store it anywhere, we will need to calculate on each new config update.
I agree it's not that clear right now, would love to address it oneday and improvre